File: eval/corpus/ts/encoder.ts
// Encoder batches values and flushes them as one frame.
export class Encoder {
  private pending: number[] = [];
  flush(): number {
    const n = this.pending.length;
    this.pending = [];
    return n;
  }
  drive(): number {
    return this.flush();
  }
}

File: eval/corpus/cpp/encoder.cpp
// Encoder batches values and flushes them as one frame.
class Encoder {
public:
    int flush();
    int drive();
private:
    int count;
};

int Encoder::flush() {
    int n = count;
    count = 0;
    return n;
}

int Encoder::drive() {
    return this->flush();
}
